code/modules/events/_event.dm

```diff
@@ -131,6 +130,19 @@
 /datum/round_event/proc/start()
 	return
 
+//Called after start()
+//Lets the ghosts know that the event has started, and provides a follow link to a mob if possible
+//Only called once.
+/datum/round_event/proc/announce_to_ghosts()
+	if(control.max_occurrences == 0) //Event will not spawn
```

Comment:
Events forced through admin tools that wouldn't happen randomly would fail this. Why even check this here ? By the point this announcement happens it should be known if the event succeeded or not ?

Reply:
Uuh a few things set max_occurrences to zero specifically to keep the event from firing. Ex, if there arent enough players, set max_occurences to zero (so that the event doesnt spawn).

Reply:
I don't see how that's relevant. If it's forced by admin these will prevent events with 0 max occurrences to announce. If fired randomly the announcement won't happen, it's fired after you know you can actually run the event.

Reply:
@AnturK 
![image](https://user-images.githubusercontent.com/29008542/47164294-c7f65500-d2c5-11e8-9c9b-1a407169c9d3.png)
Its relevent because max_occurences gets set to zero when events cant run

Reply:
Is announcement invoked from preRunEvent

Reply:
ive deduced that i might not have braincells

Reply:
Wait no. the reason for this check is because ghost role events would call start() even if they were unable to spawn. If they were unable to spawn, their max_occurences would be set to 0 (to keep them from spawning) and i incorrectly assumed that setting max_occurences to 0 was the standard for preventing an event spawn

Reply:
ghost_announce is invoked after start()